File: 2020/01/report_repair.py
# report_repair.py
# Callum Howard, 2020

import sys

def solve2(nums, target):
    for num in nums:
        if target - num in nums:
            return (num * (target - num))

def solve(nums, target):
    return next(n * (target - n) for n in nums if target - n in nums)

print(solve(set([int(n) for n in sys.stdin.readlines()]), 2020))

File: 2020/01/report_repair.go
// Report Repair
// https://adventofcode.com/2020/day/1

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	const target = 2020
	var scanner = bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanLines)

	var nums = make(map[int]bool)
	for scanner.Scan() {
		var num, _ = strconv.Atoi(scanner.Text())
		nums[num] = true
	}

	for key := range nums {
		if nums[target-key] {
			fmt.Println(key * (target - key))
			break
		}
	}
}
